### DISCUSSION TITLE
CLS와 FCP 향상과 ALT 태그 적용이 필요합니다.

### DISCUSSION THREAD

**Discussion by DjK1LLeR** [pronist/hELLO#97] [Q&A] [answered]
안녕하세요! 우선 좋은 스킨을 개발해주셔서 정말 감사합니다.
구글 콘솔 Lighthouse 에서 SEO 점수부분에서
첨부된 이미지들에 ALT 태그가 미적용된 상태로 확인됩니다.

또한 전반적으로 FCP, LCP 가 너무 오래 잡아먹습니다.

똑같은 게시글로 스킨만 바꾸어 테스트 해서 비교를 해봐도 LCP 시간이 너무 느린데

개선할수있는지 궁금합니다 .감사합니다.

**Comment by pronist** [comment, marked as answer]
`alt` 는 티스토리에서 기본적으로 붙여주어야하는게 원래 맞는 동작이지만, 붙여주지 않습니다. 심지어 티스토리에서 기본적으로 제공하는 스킨도 Lighthouse 점수는 그렇게 높게나오는 편이 아니죠. 그 외에 지표들에 대해서는 특히 **글 읽기 모드**에 있는 스크린, 테이프 와 같은 `Styled Heading` 에 의해 느려집니다. 예를 들어 글 읽기 모드가 기본적으로 `테이프` 인 경우 대표이미지가 없을 경우 `테이프` 가 아닌 일반적인 글 읽기 모드로 바뀝니다. 게다가 추가적인 기능으로 블로거가 글을 쓰면서 글 읽기 모드를 수동으로 바꿀 수도 있습니다. 이 과정에서 레이아웃을 다시 그리기 때문에 웹페이지의 로드 속도가 느려집니다. 

이 부분의 속도를 높히려면 `Styled Heading` 기능을 완전히 제거하거나 레이아웃을 다시 그리는 동작을 하지 않도록 글 읽기 모드가 자동 변경 및 수동으로 글 읽기 모드를 변경하는 기능을 날려야할 것입니다. 그렇게되면 레이아웃을 다시 재정렬하지 않고 지표가 상승합니다. 그 대신 지불해야 하는 것은 사용자의 편의성입니다.

웹페이지의 속도와 가독성 및 편의성 사이에서 선택해야 할 부분입니다. 속도가 아닌 가독성과 편의성을 선택한거죠. 속도가 중요하다면 코드블럭의 가독성 향상을 위한 별도의 폰트를 로드하지 않아야하고, 댓글접기, 카테고리 접기, 이전 글/다음 글, Scrollspy 구현을 위한 CSS 프레임워크를 로드하지 않고 직접 구현하거나 기능자체를 삭제해야 할 것입니다. 속도를 중시하고자 한다면 스킨에서 제공하는 일부 기능에 대한 제거가 필요하다는 사실을 알려드립니다.

물론 스킨을 패치하면서 속도에 대한 부분도 신경쓰려고 하고 있었고 Lighthouse 의 지표에 대해서 말씀해주신 부분 이미 인지를 하고 있는 상황이었습니다. 다만 선택의 기로에 있을 뿐이라서 여러모로 고민하고 있습니다. :D

**Reply by DjK1LLeR** [reply]
친절한 답변 감사드립니다. SEO 최적화에서는 상위 노출을 위해 Lighthouse 점수가 중요해서 , 제작하신 스킨은 제가봐도 역대급 최고 스킨임은 확실합니다.  폰트 로드만 없애도 상당히 속도향상이 있는데, 폰트 상태된 버전이나 기타 기능 삭제한 버젼도 같이 업로드 해주시면 정말 좋을거 같습니다. 늘 항상 건강하고 행복하세요!

**Reply by pronist** [reply]
@DjK1LLeR 고맙습니다. 다만, 기능을 삭제한 버전이나 폰트 로드를 제거한 버전은 게시할 계획이 없음을 알려드립니다. 저 같은 경우에는 스킨을 그대로 사용하고 있음에도 구글검색페이지에서 1페이지 또는 제일 상단에 나오는 포스팅도 제법 있기때문에 크게 문제라고 여기지는 않고 있기 때문입니다.
